app/models/subscription.rb

```diff
@@ -1,2 +1,22 @@
 class Subscription < ApplicationRecord
+  belongs_to :company
+  belongs_to :payment_method
+  belongs_to :plan
+  belongs_to :purchaser
+  enum status: { active: 0, cancelled: 10, overdue: 20 }
+  before_create :save_aniversary_day
+  before_create :generate_token
+
+  private
+
+  def generate_token
+    self.token = SecureRandom.hex(20)
+    generate_token if Product.exists?(token: token)
+  end
+
+  def save_aniversary_day
+    return self.aniversary_day = 1 if Time.zone.today.day > 28
+
+    self.aniversary_day = Time.zone.today.day
```

Comment:
Um outro gente que a gente vê muito seria algo assim:

```suggestion
    self.aniversary_day = 
      if Time.zone.today.day > 28
        1
      else
        Time.zone.today.day
      end
```

Ou usando ternário, essa é só um comentário FYI